src/_posts/2019-10-30-crafting-websites-using-finite-state-machines-part-I.md

```diff
@@ -0,0 +1,403 @@
+---
+author: Dan Bunea
+layout: post
+asset-type: post
+title: "Crafting web apps using finite state machines - Part 1"
+date: 2019-10-30 00:00:00
+description: A practical approach to crafting web applications, with the goals of breaking the complexity down (state machines, separation of concerns) and building robustness right into the system (atomicity with rollback).
+image:
+    src: /assets/custom/img/blog/2019-10-30-crafting-websites-using-finite-state-machines-part-I/crafting/controller%20to%20diagram.png
+    attribution:
+
+abstract: Crafting web apps using finite state machines, separation of concerns and atomicity.
+tags:
+- functional programming
+- ClojureScript
+- finite state machines
+- model view controller
+---
+
+# Crafting web apps using finite state machines - Part 1
+
+
+[Code on Gitlab](https://gitlab.com/danbunea/how-to-solve-it-crafting-web-apps)
+
+**Table of contents**
+
+- PROBLEM: How should we start crafting a web app?
+- SOLUTION
+  * THEORY
+  * Strategy
+    + Goals
+  * Tactics
+    + Patterns
+    + Techniques
+  * PRACTICE
+    + Step 1 Draw the diagram of the UI state machine
+    + Step 2: Domain modelling
+    + Step 3: Solve the data problem
+    + Step 4: Build the UI
+  * Goals revised
+  * Part II - what will be discussed
```

Comment:
Probably this should be one level to the left?

Reply:
Done